src/EnergyPlus/ZoneEquipmentManager.cc

```diff
@@ -4328,25 +4319,75 @@ namespace EnergyPlus::ZoneEquipmentManager {
+                        } else {
+                            returnNodeMassFlow = 0.0;
+                        }
                     }
                 }
+                zoneTotReturnFlow += returnNodeMassFlow;
+            }
+            // Adjust return node flows if zone total return flow is > 0
+            if (zoneTotReturnFlow > 0.0) {
+                for (int returnNum = 1; returnNum <= numRetNodes; ++returnNum) {
+                    int retNode = thisZoneEquip.ReturnNode(returnNum);
+                    if (retNode > 0) {
+                        if (numRetNodes == 1) {
+                            // set it to expected return flows
+                            DataLoopNode::Node(retNode).MassFlowRate = ExpTotalReturnMassFlow;
+                            FinalTotalReturnMassFlow = ExpTotalReturnMassFlow;
+                        } else { // multiple return nodes, adjust nodes flow
+                            Real64 newReturnFlow = 0.0;
+                            Real64 returnAdjFactor = ExpTotalReturnMassFlow / zoneTotReturnFlow;
+                            Real64 curReturnFlow = DataLoopNode::Node(retNode).MassFlowRate;
+                            newReturnFlow = curReturnFlow * returnAdjFactor;
+                            DataLoopNode::Node(retNode).MassFlowRate = newReturnFlow;
+                            FinalTotalReturnMassFlow += newReturnFlow;
+                        }
+                    }
+                } 
+            } else {
+                FinalTotalReturnMassFlow = ExpTotalReturnMassFlow;
             }
+
         } else {
-            FinalTotalReturnMassFlow = totReturnFlow;
+
+            // Adjust return flows if greater than expected (i.e. there is exhaust or mixing flow reducing the total available for return)
+            if ((totReturnFlow > ExpTotalReturnMassFlow) && (totVarReturnFlow > 0.0)) {
+                Real64 newReturnFlow = 0.0;
+                Real64 returnAdjFactor = (1 - ((totReturnFlow - ExpTotalReturnMassFlow) / totVarReturnFlow)); // Return flow adjustment factor
+                for (int returnNum = 1; returnNum <= numRetNodes; ++returnNum) {
+                    int retNode = thisZoneEquip.ReturnNode(returnNum);
+                    Real64 curReturnFlow = DataLoopNode::Node(retNode).MassFlowRate;
+                    if (retNode > 0) {
+                        if (!thisZoneEquip.FixedReturnFlow(returnNum)) {
+                            newReturnFlow = curReturnFlow * returnAdjFactor;
+                            FinalTotalReturnMassFlow += newReturnFlow;
+                            DataLoopNode::Node(retNode).MassFlowRate = newReturnFlow;
+                        } else {
+                            FinalTotalReturnMassFlow += curReturnFlow;
+                        }
+                    }
+                }
+            } else {
+                FinalTotalReturnMassFlow = totReturnFlow;
+            }
```

Comment:
Github mixed up the existing and the new code section. It better to see the changes on actual source code. I have used IF- ELSE to separate the new and the existing methods for adjusting the zone return node flow rates.